packages/file_selector/file_selector_android/android/src/main/java/dev/flutter/packages/file_selector_android/FileUtils.java

```diff
@@ -130,11 +130,14 @@ public static String getPathFromCopyOfFileFromUri(@NonNull Context context, @Non
         fileName = getBaseName(fileName) + extension;
       }
 
-      File file = new File(targetDirectory, fileName);
+      String filePath = new File(targetDirectory, fileName).getPath();
```

Comment:
I'm pretty sure you want to sanitize before you pass it to `File`, since once you've attached the `targetDirectory` you are stripping off information we've added during sanitization.

Reply:
I believe the sanitization should have already happened here, as we get these values earlier:
```java
      String fileName = getFileName(context, uri);
      String extension = getFileExtension(context, uri);
```

And I added the sanitization to each of those methods. Does that make sense?

Reply:
I see; sorry, I misread in the initial readthrough.